Doc/reference/datamodel.rst

```diff
@@ -1946,10 +1946,12 @@ Once the appropriate metaclass has been identified, then the class namespace
 is prepared. If the metaclass has a ``__prepare__`` attribute, it is called
 as ``namespace = metaclass.__prepare__(name, bases, **kwds)`` (where the
 additional keyword arguments, if any, come from the class definition). The
-``__prepare__`` method should be implemented as a :func:`classmethod`.
+``__prepare__`` method should be implemented as a :func:`classmethod`. The
+namespace returned by ``__prepare__`` is passed in to ``__new__``, but when
+the final class object is created the namespace is copied into a new ``dict``.
 
 If the metaclass has no ``__prepare__`` attribute, then the class namespace
-is initialised as an empty ordered mapping.
```

Comment:
Per PEP 520, the default class namespace must be an ordered mapping and the original text here was purposefully phrased.  While we've talked about requiring "dict" to be ordered ([in the language reference](https://docs.python.org/3/reference/datamodel.html#index-30)), that hasn't happened.  So changing the language spec text here to "dict" is inappropriate without a PEP.  Note that [elsewhere in the language reference](https://docs.python.org/3/reference/datamodel.html#creating-the-class-object) the "ordered mapping" phrasing still exists.

Reply:
Um, `dict` *is* required to be ordered. Since 3.7. See the 3.7. [what's new](https://docs.python.org/3/whatsnew/3.7.html) doc "the insertion-order preservation nature of dict objects has been declared to be an official part of the Python language spec."

If after that has sunk in you still think this PR is wrong, please submit a new PR to fix it! (Sorry for landing prematurely.)

Reply:
Per my comment on the bpo I think @ericsnowcurrently is right but for the wrong reasons.   I have opened a PR correcting the wording.  https://github.com/python/cpython/pull/18682